src/ucs/debug/log.c

```diff
@@ -355,6 +356,7 @@ void ucs_log_cleanup()
     if (ucs_log_file_close) {
         fclose(ucs_log_file);
     }
+    ucs_spinlock_destroy(&threads_lock);
```

Comment:
probably shouldn't use spinlock like that from logger, because ucs_spinlock uses the logger itself..

Reply:
Revert use `ucs_spinlock`. Use `pthread_spinlock` in `src/ucs/debug/log.c`
Additionally, I add `pthread_spin_destroy` in the same file.
Please take a look when you get a chance.